### PR TITLE
Modify Environment on TravisCI

### PR DESCRIPTION
**Pull request by Coderlane**
This adds a section to run-standard-tests.sh to configure various
environment settings on TravisCI hosts. Currently we'll do the following:

1) On Linux hosts re-enable IPv6 support.
2) On OSX hosts increase the global maximum open file descriptor limit to 8192.

This is the successor to https://github.com/libp2p/go-reuseport/pull/35. I [tested the script](https://travis-ci.org/Coderlane/go-reuseport/builds/328503024) and [another modification](https://github.com/libp2p/go-reuseport/commit/b82cb3a4b0ff85d7a85632c45b020f32a8a3565b) on TravisCI and it seems to fix the issues go-reuseport was having on OSX.

### CODE REVIEW

**Comment by Stebalien** [comment]
Awesome! One change: this will need to check `TRAVIS_SUDO` to see if we have it enabled (we don't enable it on all hosts).

**Comment by Coderlane** [comment]
Good catch, I've added that check. I also limited the ulimit setting to OSX, because it the Linux runs don't need it.

**Comment by Stebalien** [comment]
Thanks!

File: travis-ci/run-standard-tests.sh [hunk 1]
```diff
@@ -25,6 +25,25 @@ echo
...
+        sudo sysctl -w net.ipv6.conf.lo.disable_ipv6=0
+        sudo sysctl -w net.ipv6.conf.default.disable_ipv6=0
+        sudo sysctl -w net.ipv6.conf.all.disable_ipv6=0
+    fi
+else
+    if [[ "$TRAVIS_SUDO" == true ]]; then
+        # OSX has a default file limit of 256, for some tests we need a
+        # maximum of 8192.
+        sudo launchctl limit maxfiles 8192 8192
+        ulimit -n 8192
```
Comment (kevina) [inline travis-ci/run-standard-tests.sh]: I am not sure why `sudo launchctl limit maxfiles 8192 8192` is needed but I would think `ulimit -n 8192` should always be used on mac osx, unless it does not do anything without `sudo launchctl limit maxfiles 8192 8192`.

Comment (Coderlane) [reply travis-ci/run-standard-tests.sh]: I was assuming that I needed to change the global maximum, but it looks like it is already set at unlimited. The launchctl limit line is apparently unnecessary, and ulimit could be used alone.

Comment (kevina) [reply travis-ci/run-standard-tests.sh]: @Coderlane do you want to create another P.R. to remove the sudo check for OsX and just use `ulimit -n 8192`?

**Pull request merged**